core/src/test/scala/org/apache/spark/deploy/IvyTestUtils.scala

```diff
@@ -256,6 +320,7 @@ private[deploy] object IvyTestUtils {
       } else {
         FileUtils.deleteDirectory(repo)
       }
+      FileUtils.deleteDirectory(new File(ivySettings.getDefaultCache, artifact.groupId))
```

Comment:
We need to delete the cache for these fake libraries, or they affect each other between tests.